js/components/graph/Graph.js

```diff
@@ -973,7 +1007,10 @@ export class Graph extends React.Component {
       this.setState(
         prevState => {
           if (nodeId in prevState.nodesStatus) {
-            const nodesStatus = { ...prevState.nodesStatus }
+            const nodesStatus = {}
+            Object.keys(prevState.nodesStatus).forEach(key => {
```

Comment:
I thought it would be better in terms of signaling React to re-render, but after reading and thinking a bit I do not think it makes a difference. It suffices to copy the top-level for signaling React to re-render.